network/src/peer_manager/mod.rs

```diff
@@ -459,50 +459,63 @@ where
-                if conn.metadata.origin == ConnectionOrigin::Outbound {
-                    let peer_role = self
-                        .trusted_peers
-                        .read()
-                        .get(&conn.metadata.remote_peer_id)
-                        .map_or(PeerRole::Unknown, |auth_context| auth_context.role);
-                    conn.metadata.role = peer_role;
-                };
-
-                // TODO: Keep track of somewhere else to not take this hit in case of DDoS
-                let inbound_conns = self
-                    .active_peers
-                    .iter()
-                    .filter(|(_, (metadata, _))| metadata.origin == ConnectionOrigin::Inbound)
-                    .count();
-
-                // Reject excessive inbound connections by letting them just drop out of scope
-                // We control outbound connections with Connectivity manager before we even send them
-                // and we must allow connections that already exist to pass through tie breaking.
-                // TODO: Allow for trusted peers to still connect
-                if conn.metadata.origin == ConnectionOrigin::Outbound
-                    || self
-                        .active_peers
-                        .contains_key(&conn.metadata.remote_peer_id)
-                    || inbound_conns < self.inbound_connection_limit
-                {
-                    info!(
-                        NetworkSchema::new(&self.network_context)
-                            .connection_metadata_with_address(&conn.metadata),
-                        "{} New connection established: {}", self.network_context, conn.metadata
-                    );
-                    // Add new peer, updating counters and all
-                    self.add_peer(conn);
-                    self.update_connected_peers_metrics();
-                } else {
-                    info!(
-                        NetworkSchema::new(&self.network_context)
-                            .connection_metadata_with_address(&conn.metadata),
-                        "{} Connection rejected due to connection limit: {}",
-                        self.network_context,
-                        conn.metadata
-                    );
-                    self.disconnect(conn);
+                let trusted_peers = self.trusted_peers.read().clone();
+
+                match conn.metadata.origin {
+                    ConnectionOrigin::Outbound => {
+                        // TODO: This is right now a hack around having to feed trusted peers deeper in the outbound path.  Inbound ones are assigned at Noise handshake time.
+                        conn.metadata.role = trusted_peers
+                            .get(&conn.metadata.remote_peer_id)
+                            .map_or(PeerRole::Unknown, |auth_context| auth_context.role);
```

Comment:
actually I think this okay, in fact, I would like to log the case where we  have an unknown outbound, that sounds like a nice bad thing happening.